File: Api/src/Api.Domain/Entities/TeamEntity.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities {
    public class TeamEntity : BaseEntity
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }
        
        [Required]
        [MaxLength(1000)]
        public string Description { get; set; }

        public IEnumerable<TeamProjectEntity> TeamProjects { get; set; }

        public IEnumerable<UserTeamEntity> UserTeams { get; set; }

        [Required]
        [MaxLength(7)]
        public string Color { get; set; }
    }
}

File: App/src/app/team/entities/team.entity.ts
import { BaseEntity } from '../../shared/entities/base.entity';
import { TeamProjectEntity } from '../../shared/entities/teamProject.entity';
import { UserTeamEntity } from '../../shared/entities/userTeam.entity';

export interface TeamEntity extends BaseEntity {
  name: string;
  description: string;
  teamProjects?: TeamProjectEntity[];
  userTeams?: UserTeamEntity[];
  color: string;
}
